### ISSUE TITLE
RPC methods missing test coverage 

### ISSUE THREAD

**Issue by patricklodder** [dogecoin/dogecoin#1640]
To defend against regressions introduced by new code, good coverage of RPC commands in `qa/rpc-tests` is desirable.

As of this writing, 1.14.3-dev misses coverage for the commands listed below.

##### More relevant 
- [ ] bumpfee
- [ ] estimatefee
- [ ] getaddednodeinfo
- [x] getblockchaininfo - #1637
- [ ] getconnectioncount
- [ ] getdifficulty
- [ ] getinfo
- [ ] getmemoryinfo
- [ ] getmempoolancestors
- [ ] getmempooldescendants
- [ ] getmempoolentry
- [ ] getmininginfo
- [ ] getnettotals
- [ ] getnetworkhashps
- [ ] gettxout
- [ ] pruneblockchain
- [ ] setnetworkactive

##### Less relevant
- [ ] addwitnessaddress
- [ ] estimatepriority
- [ ] estimatesmartfee
- [ ] estimatesmartpriority
- [ ] listaddressgroupings
- [ ] ping

**Comment by gewure** [comment]
we should label testing issues

**Comment by gewure** [comment]
there are some corresponding python files in 1.14.3-dev though? 
https://github.com/dogecoin/dogecoin/blob/1.14.3-dev/qa/rpc-tests/bumpfee.py

**Comment by patricklodder** [comment]
Correct, but that test has been disabled from pull-tester execution because it wasn't ported yet: 

https://github.com/dogecoin/dogecoin/blob/0b46a40ed125d7bf4b5a485b91350bc8bdc48fc8/qa/pull-tester/rpc-tests.py#L156-L158

So work still needs to get done on it.

**Comment by gewure** [comment]
So what exxactly needs to be done?

**Comment by ReverseControl** [comment]
@patricklodder What is the status of this? Are we adding it to 1.14.5?

**Comment by gewure** [comment]
> @patricklodder What is the status of this? Are we adding it to 1.14.5?

Its open and not ported. Imo a big ticket, a lot of python code and rather deep knowledge of the RPC-API needed.

**Comment by patricklodder** [comment]
We've made some progress on this with making `bumpfee.py` and `p2p-acceptblock.py` working, but we're far from there.

At the moment I am more worried about the protocol tests (like `p2p-compactblocks.py`) not working than this, but any improvements that tick things off this list are awesome - don't have to be done all at once.

**Comment by gewure** [comment]
Wouldn't it make sense to split this up into more digestable tickets, some of them are much more effort than others.

unfortunately im neither good in python nor in coindaemon RPCs - else i would have done something here already but the effort to comprehend seems quite high.

**Comment by patricklodder** [comment]
This was mostly just meant to be a list that reflects which RPC methods have no test at all.

Some of those should basically just be contract tests so that we can track breaking changes to the API, i.e. gettxout can be put into a test that creates some outputs and then queries the RPC to make sure that:

a) the outputs are listed
b) the fields are showing values as expected

and done.

I can create an example PR for one or two of these - would that help?

**Comment by ReverseControl** [comment]
@patricklodder Examples always help.
